scripts/core/counterbasedeventpool.yml

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (C) 2019-2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Extension APIs for Counter-based Event Pools"
+version: "1.8"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Counter-based Event Pools Extension Name"
+version: "1.4"
+name: $X_EVENT_POOL_COUNTER_BASED_EXP_NAME
+value: '"$X_experimental_event_pool_counter_based"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Counter-based Event Pools Extension Version(s)"
+version: "1.8"
+name: $x_event_pool_counter_based_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported event flags for defining counter-based event pools."
+class: $xEventPool
+name: $x_event_pool_counter_based_exp_flags_t
+etors:
+    - name: IMMEDIATE
+      desc: "Counter-based event pool is used for immediate command lists (default)"
+    - name: NON_IMMEDIATE
+      desc: "Counter-based event pool is for non-immediate command lists"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Event pool descriptor for counter-based events. This structure may be passed to $xEventPoolCreate as pNext member of $x_event_pool_desc_t."
+class: $xEventPool
+version: "1.8"
+name: $x_event_pool_counter_based_exp_desc_t
+base: $x_base_desc_t
+members:
+    - type: $x_event_pool_counter_based_exp_flags_t
+      name: flags
+      desc: |
+            [in] mode flags.
+            must be 0 (default) or a valid value of $x_event_pool_counter_based_exp_flag_t
+            default behavior is counter-based event pool is only used for immediate command lists.
+      init: "0"
```

Comment:
do we need member for `count` which is num events in the pool? we already have a count in `event_pool_desc_t` so are we going to just use that? just making sure so that we document this in a pseudo-code clearly

Reply:
count still comes from vent_pool_desc_t. here we are only adding a flag.